bitmark-references/bitmark-node-software/bitmark-blockain.md

```diff
@@ -0,0 +1,125 @@
+1. Incoming block difficulty is same as current difficulty
+1. Incoming block hash meets difficulty criteria
+1. Incoming block's previous block hash same as current block hash
+1. Incoming transaction fits each type's format and length
+1. Incoming transaction owner ownership is valid
+1. Incoming transaction payment is valid
+1. Incoming block's sha3 of all transactions inside is valid
+
+After a block is verified, a node stores that block into its internal
+database (leveldb), then node broadcast this newly saved block to all connected
+nodes. When other node receives the broadcast of a block, each node
+verifies block correctness by rules described above.
+
+### Block Synchronization
+
+Bitmark blockchain uses both proof-of-work (PoW) and majority votes to
+choose blocks.
+
+#### Proof of Work
+
+A proof of work is a piece of data which is difficult (costly, time-consuming) to produce but easy for others to verify and which satisfies certain requirements. Producing a proof of work can be a random process with low probability so that a lot of trial and error is required on average before a valid proof of work is generated. Bitcoin uses the Hashcash proof of work system.
+
+In order for a block to be accepted by network participants, miners must complete a proof of work which covers all of the data in the block. The difficulty of this work is adjusted so as to limit the rate at which new blocks can be generated by the network to one every 2 minutes. Due to the very low probability of successful generation, this makes it unpredictable which worker computer in the network will be able to generate the next block.
+
+For a block to be valid it must hash to a value less than the current target; this means that each block indicates that work has been done generating it. Each block contains the hash of the preceding block, thus each block has a chain of blocks that together contain a large amount of work. Changing a block (which can only be done by making a new block containing the same predecessor) requires regenerating all successors and redoing the work they contain. This protects the block chain from tampering.
+
+Bitmark uses [argon2](https://en.wikipedia.org/wiki/Argon2) to hash the block header; remaining data is covered by having the Merkle Tree Root hash as part of the header. If transactions are different or in a different order the the Merkle Root has will not match and the block will be rejected.
+
+#### Majority Votes
+
+When deciding next block, every node considers decisions from other nodes, the most block selected by other nodes will be chosen. Every 30 seconds, a node asks its connected nodes for their block hash and block height. This information will be used to determine majority votes. For a specific chain, same block hegith exists same block hash, so a node can know itself status by this information.
+
+Some conditionas could happen when deciding block:
+
+1. majority votes exists
+
```

Comment:
It's just slightly confusing here having a hash of "abcdefg" when your node names are A, B, C, D, E, F, and later G.

Reply:
update description